### PR TITLE
chore: move to app router, improve performance (HNT-6117)

### PR DESCRIPTION
**Pull request by miguel-nascimento**
app router will give us more control on what we can cache

page router: https://pagespeed.web.dev/analysis/https-river-build-delegate/vb4o2nhbrp?form_factor=mobile
app router: https://pagespeed.web.dev/analysis/https-river-build-www-git-miguel-app-router-river-build-team-vercel-app-delegate/fubxg5im4n?form_factor=mobile

### CODE REVIEW

File: src/app/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+import { GeistSans } from 'geist/font/sans'
+import { Metadata } from 'next'
+import localFont from 'next/font/local'
+
+// Font files can be colocated inside of `pages`
+const menloFont = localFont({ src: '../lib/fonts/Menlo-Regular.woff', variable: '--font-menlo' })
+
+const DESCRIPTION =
+  'River is a permissionless protocol for social networking apps. Create dynamic on-chain spaces that feature memberships, reputation, and end-to-end message encryption.'
+const metaImageUrl = 'https://hnt-river-dev-portal.vercel.app/og-image.jpeg'
```
Comment (jterzis) [inline src/app/layout.tsx]: Can we read this static asset from source instead of a url that may break or change ?

File: package.json [hunk 2]
```diff
@@ -48,10 +48,10 @@
     "lucide-react": "^0.236.0",
     "mini-svg-data-uri": "^1.4.4",
     "mixpanel-browser": "^2.45.0",
-    "next": "^13.2.4",
-    "react": "^18.2.0",
+    "next": "^14.2.3",
```
Comment (jterzis) [inline package.json:51]: Is there a minimum node version we need with next14 ?

File: src/app/layout.tsx [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+import { siteDataQuery } from '@/gql/query'
+
+import { GeistSans } from 'geist/font/sans'
+import { Metadata } from 'next'
+import localFont from 'next/font/local'
+
+// Font files can be colocated inside of `pages`
+const menloFont = localFont({ src: '../lib/fonts/Menlo-Regular.woff', variable: '--font-menlo' })
+
+const DESCRIPTION =
```
Comment (jterzis) [inline src/app/layout.tsx:14]: Did we have to move the layout homepage to the app router as well as part of this optimization for loading delegate page ?

File: package.json (hunk 2 above)
Comment (miguel-nascimento) [reply package.json:51]: Node 18, which is the one that we have on package.json

File: src/app/layout.tsx (hunk 3 above)
Comment (miguel-nascimento) [reply src/app/layout.tsx:14]: We need to move at least the `_app.ts` (page router) to `layout.tsx` (app router) but the page itself I can move it back to the page router

File: src/app/layout.tsx (hunk 1 above)
Comment (miguel-nascimento) [reply src/app/layout.tsx]: Nice suggestion, done!

**Review by jterzis — APPROVED** [review]

**Pull request merged**